templates/ci/docker.yml

```diff
@@ -0,0 +1,63 @@
+# Docker
+# Build, tag, push, or run Docker images, or run a Docker command.
+# More Information on https://go.microsoft.com/fwlink/?linkid=848006
+# YAML help: https://aka.ms/yaml
+
+queue:
+  name: Hosted VS2017                                       # Hosted Linux Preview, Default
+
+variables:
+#  system.debug: true
+
+  dockerRegistryConnectionEndpoint: ''                      # The unique name or the GUID Id of VSTS service connection endpoint
+                                                            # for Docker registry. Required if the step needs to authenticate
+                                                            # with a registry.
+                                                            
+  dockerHostEndpoint: ''                                    # The unique name or the GUID Id of VSTS service connection endpoint
+                                                            # for Docker host. Defaults to the agent's host.
+                                                            
+  imageName: '$(Build.Repository.Name):$(Build.BuildId)'    # Name of the Docker image to build and push
+  
+  qualifyImageName: true                                    # Qualify the image name with the Docker registry connection's hostname 
+                                                            
+  enforceDockerNamingConvention: true                       # The docker image name will be modified to follow Docker naming
+                                                            # convention. Converts upper case character to lower case and
+                                                            # removes spaces in image name.
+
+  defaultContext: true                                      # Set the build context to the directory that contains the Docker file.
+
+steps:
+- task: Docker@0
+  inputs:
+    containerregistrytype: 'Container Registry'
```

Comment:
This seems very verbose to include. Can we limit the number of inputs to a smaller number and rely on defaults being smart. Can we limit to containerregistrytype, dockerRegistryConnection, action and dockerFile?

Reply:
When I "View YAML" of the "Container" template in the designer, the YAML looks like this:

resources:
- repo: self
queue:
  name: Hosted VS2017
steps:
- task: Docker@0

- task: Docker@0
  inputs:
    action: 'Push an image'

Reply:
I'll double check, but that looks like a bug in the `View YAML`.

Reply:
*> I see the template is called "Container" in the UI...*
That makes sense for the Designer template. It supports two types of containers, i.e. Docker and Azure, and any combinations of them. This template represents only of for combinations - Docker 
 for both tasks in the template. IMO, we can use the `condition` feature of the YAML process and implement both options, but IMO that's too much for the first implementation.

Reply:
*> This seems very verbose...*
As a user, I prefer the more verbose templates. A reasonable amount of extra comments as well as default and other available values provided does not affect the speed in production, but simplifies the development process. Let's discuss that offline.